src/main/java/mate/academy/springboot/swagger/conroller/ProductController.java

```diff
@@ -0,0 +1,93 @@
+package mate.academy.springboot.swagger.conroller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.AllArgsConstructor;
+import mate.academy.springboot.swagger.dto.ProductRequestDto;
+import mate.academy.springboot.swagger.dto.ProductResponseDto;
+import mate.academy.springboot.swagger.mapper.DtoMapper;
+import mate.academy.springboot.swagger.model.Product;
+import mate.academy.springboot.swagger.service.PaginationService;
+import mate.academy.springboot.swagger.service.ProductService;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@AllArgsConstructor
+@RestController
+@RequestMapping("/products")
+public class ProductController {
+    private final ProductService productService;
+    private final DtoMapper<Product, ProductRequestDto, ProductResponseDto> dtoMapper;
+    private final PaginationService paginationService;
+
+    @Operation(summary = "Add product to DB")
+    @PostMapping
+    public ProductResponseDto create(@RequestBody ProductRequestDto dto) {
```

Comment:
Let's add a description schema of a request DTO's for each endpoint where we have it 
https://stackoverflow.com/questions/64645528/java-spring-boot-openapi-3-how-to-add-description-for-requestbody